Source/SwiftLintFramework/Rules/Lint/UnusedImportRuleExamples.swift

```diff
@@ -0,0 +1,192 @@
+struct UnusedImportRuleExamples {
+    static let nonTriggeringExamples = [
+        Example("""
+        import Dispatch // This is used
+        dispatchMain()
+        """),
+        Example("""
+        @testable import Dispatch
+        dispatchMain()
+        """),
+        Example("""
+        import Foundation
+        @objc
+        class A {}
+        """),
+        Example("""
+        import UnknownModule
+        func foo(error: Swift.Error) {}
+        """),
+        Example("""
+        import Foundation
+        import ObjectiveC
+        let 👨‍👩‍👧‍👦 = #selector(NSArray.contains(_:))
+        👨‍👩‍👧‍👦 == 👨‍👩‍👧‍👦
+        """)
+    ]
+
+    static let triggeringExamples = [
+        Example("""
+        ↓import Dispatch
+        struct A {
+          static func dispatchMain() {}
+        }
+        A.dispatchMain()
+        """),
+        Example("""
+        ↓import Foundation // This is unused
+        struct A {
+          static func dispatchMain() {}
+        }
+        A.dispatchMain()
+        ↓import Dispatch
+
+        """),
+        Example("""
+        ↓import Foundation
+        dispatchMain()
+        """),
+        Example("""
+        ↓import Foundation
+        // @objc
+        class A {}
+        """),
+        Example("""
+        ↓import Foundation
+        import UnknownModule
+        func foo(error: Swift.Error) {}
+        """),
+        Example("""
+        ↓import Foundation
+        typealias Foo = CFData
+        """, configuration: [
+            "require_explicit_imports": true
+        ])
```

Comment:
I already wrote it somewhere. there' s an idea to add a comment to the Example structure so that
it will explain why trigger should be triggered :) Totally out of the scope of this PR
```
Example("""
        ↓import Foundation
        typealias Foo = CFData
        """, configuration: [
            "require_explicit_imports": true
        ],comment: "Using transitive type 'CFData' from CoreFoundation should require CoreFoundation import")
```

This example is pretty obvious, but there are some examples that will help understanding why one or another rule should or should not be triggered.

also, this comment part can be shown in violation. I think additional PR will be needed :)

Reply:
Ah yeah, that would be great for documentation purposes, but also to have nicer descriptions when tests fail.

Reply:
Added. as an issue here. 
https://github.com/realm/SwiftLint/issues/3125